src/main/java/org/folio/circulation/domain/Loan.java

```diff
@@ -220,37 +229,61 @@ public User getProxy() {
 
   Loan withProxy(User newProxy) {
     return new Loan(representation, item, user, newProxy, checkinServicePoint,
-      checkoutServicePoint, originalDueDate);
+      checkoutServicePoint, originalDueDate, loanPolicy);
   }
-  
+
   public Loan withCheckinServicePoint(ServicePoint newCheckinServicePoint) {
     return new Loan(representation, item, user, proxy, newCheckinServicePoint,
-      checkoutServicePoint, originalDueDate);
+      checkoutServicePoint, originalDueDate, loanPolicy);
   }
-  
+
   public Loan withCheckoutServicePoint(ServicePoint newCheckoutServicePoint) {
     return new Loan(representation, item, user, proxy, checkinServicePoint,
-      newCheckoutServicePoint, originalDueDate);
+      newCheckoutServicePoint, originalDueDate, loanPolicy);
   }
 
-  private void changeLoanPolicy(String newLoanPolicyId) {
+  public String getLoanPolicyId() {
+    return representation.getString("loanPolicyId");
+  }
+
+  public LoanPolicy getLoanPolicy() {
+    return loanPolicy;
+  }
+
+  public Loan withLoanPolicy(LoanPolicy newloanPolicy) {
+
+    return new Loan(representation, item, user, proxy, checkinServicePoint, checkoutServicePoint, originalDueDate, newloanPolicy);
+  }
+
+  private void setLoanPolicyId(String newLoanPolicyId) {
     if (newLoanPolicyId != null) {
       representation.put("loanPolicyId", newLoanPolicyId);
     }
   }
-  
+
+  private void setLoanPolicyName(String newLoanPolicyName) {
+
+    JsonObject loanPolicySummary = representation.containsKey(LoanProperties.LOAN_POLICY)
+            ? representation.getJsonObject(LoanProperties.LOAN_POLICY)
+            : new JsonObject();
+
+    loanPolicySummary.put("loanPolicyName", newLoanPolicyName);
+
+    representation.put(LoanProperties.LOAN_POLICY, loanPolicySummary);
+  }
+
   public ServicePoint getCheckinServicePoint() {
     return this.checkinServicePoint;
   }
-  
+
   public ServicePoint getCheckoutServicePoint() {
     return this.checkoutServicePoint;
   }
 
   public Loan renew(DateTime dueDate, String basedUponLoanPolicyId) {
     changeAction("renewed");
     removeActionComment();
-    changeLoanPolicy(basedUponLoanPolicyId);
+    setLoanPolicyId(basedUponLoanPolicyId);
```

Comment:
Alternatively, this could be changed to accept a `LoanPolicy` class, rather than the raw ID now that we need more information.

Reply:
I could be, but then additional refactoring will be needed in terms of inroducing a builder for LoanPolicy

Reply:
Why would that need a builder introducing for the loan policy?

All of the places where `setLoanPolicyId` are called from appear to already have a loan policy object available. What have I missed?

Reply:
This causes test failures. Can I suggest to create a separate ticket for code enchacements?

Reply:
Yes, I tried this out yesterday and it requires some broader and delicate changes. I will raise a follow up issue, as it goes beyond the work for this change